databuilder/query_language.py

```diff
@@ -110,6 +110,9 @@ def is_in(self, other):
         # The query model requires an immutable Set type for containment queries, but
         # that's a bit user-unfriendly so we accept other types here and convert them
         if isinstance(other, (tuple, list, set)):
+            # Parse any ISO date strings in collection
```

Comment:
This will need to be implemented as a new `is_in()` method on the `DateFunctions` class rather than on `BaseSeries` because it applies only to date-typed series. As it stands, this parsing gets applied to series of any type which means that if you try to write:
```py
some_string_series == "2000-01-01"
```
you'll get a type error because it will transform the string into a date.

Reply:
but it only passes to parse_date_if_str if it's a collection, and parse_date_if_str returns the input unmodified if it isn't an iso date string format; or am I missing your point here?

Reply:
:facepalm: Sorry, I messed up the example. I meant to write:
```py
some_string_series.is_in(["2000-01-01"])
```

Reply:
gotcha